api/envoy/service/ext_proc/v3/external_processor.proto

```diff
@@ -275,8 +279,11 @@ message CommonResponse {
 
   // Replace the body of the last message sent to the remote server on this
   // stream. If responding to an HttpBody request, simply replace or clear
-  // the body chunk that was sent with that request. Body mutations only take
-  // effect in response to ``body`` messages and are ignored otherwise.
+  // the body chunk that was sent with that request. Body mutations may take
+  // effect in response either to ``header`` or ``body`` messages. When it is
```

Comment:
These "header" or "body" messages are actually from Envoy, not from external processing server.  Do you mean change into:

"When it is effect in response <from external processing server>  either to ``header`` or ``body`` messages." But I feel that is not necessary since response is always from ext_proc server.

BTW, there is no change in this part of text.  

The part changed is "and are ignored otherwise", which is wrong,  since body_mutation can happen when responding to header messages with STATUS set to CONTINUE_AND_REPLACE.  Please check the STATUS  doc and the code for more details.

Reply:
For example, the body mutation at [handleHeadersResponse](https://github.com/envoyproxy/envoy/blob/158b72649ad1b999906741bdd27c3c0a080c5e8a/source/extensions/filters/http/ext_proc/processor_state.cc#L109)  is called at [onReceiveMessage](https://github.com/envoyproxy/envoy/blob/158b72649ad1b999906741bdd27c3c0a080c5e8a/source/extensions/filters/http/ext_proc/ext_proc.cc#L599) when ext_proc receives response from external processing server. And it is based on [ProcessingResponse](https://github.com/envoyproxy/envoy/blob/158b72649ad1b999906741bdd27c3c0a080c5e8a/api/envoy/service/ext_proc/v3/external_processor.proto#L126) like response case [here](https://github.com/envoyproxy/envoy/blob/158b72649ad1b999906741bdd27c3c0a080c5e8a/source/extensions/filters/http/ext_proc/ext_proc.cc#L626).

Thus, I think that body mutation above happens when ext_proc received `ProcessingResponse` with `kRequestHeaders`/`kResponseHeader` from external server  i.e., in response to external server's message for header response case.

Reply:
Yes, that's correct. 

So, the problem before is that it states the body_mutation only take effect when responding to body request, which is wrong. It actually take effect to header request as well.

Reply:
Then what I suggested above is this body mutation is in response to message from external processing server, which is also correct?